pgmpy/inference/ExactInference.py

```diff
@@ -248,134 +248,104 @@ def query(
 
-        joint: boolean (default: True)
-            If True, returns a Joint Distribution over `variables`.
-            If False, returns a dict of distributions over each of the `variables`.
+    joint: boolean (default: True)
+        If True, returns a Joint Distribution over `variables`.
 
-        show_progress: boolean
-            If True, shows a progress bar.
+    show_progress: boolean (default: True)
+        If True, shows a progress bar.
+    """
 
-        Examples
-        --------
-        >>> from pgmpy.inference import VariableElimination
-        >>> from pgmpy.models import DiscreteBayesianNetwork
-        >>> import numpy as np
-        >>> import pandas as pd
-        >>> values = pd.DataFrame(np.random.randint(low=0, high=2, size=(1000, 5)),
-        ...                       columns=['A', 'B', 'C', 'D', 'E'])
-        >>> model = DiscreteBayesianNetwork([('A', 'B'), ('C', 'B'), ('C', 'D'), ('B', 'E')])
-        >>> model.fit(values)
-        >>> inference = VariableElimination(model)
-        >>> phi_query = inference.query(['A', 'B'])
-        """
-        evidence = evidence if evidence is not None else dict()
+    evidence = evidence if evidence is not None else dict()
 
-        # Step 1: Parameter Checks
-        common_vars = set(evidence if evidence is not None else []).intersection(
-            set(variables)
+    # Step 1: Parameter Checks
+    common_vars = set(evidence if evidence is not None else []).intersection(set(variables))
+    if common_vars:
+        raise ValueError(
+            f"Can't have the same variables in both `variables` and `evidence`. Found in both: {common_vars}"
         )
-        if common_vars:
-            raise ValueError(
-                f"Can't have the same variables in both `variables` and `evidence`. Found in both: {common_vars}"
-            )
 
-        if not variables:
-            raise ValueError(
-                "The `variables` argument to query() must contain at least one variable."
-            )
+    if not variables:
+        raise ValueError(
+            "The `variables` argument to query() must contain at least one variable."
+        )
 
-        # Step 2: If virtual_evidence is provided, modify the network.
-        if isinstance(self.model, DiscreteBayesianNetwork) and (
-            virtual_evidence is not None
-        ):
-            self._virtual_evidence(virtual_evidence)
-            virt_evidence = {"__" + cpd.variables[0]: 0 for cpd in virtual_evidence}
-            return self.query(
-                variables=variables,
-                evidence={**evidence, **virt_evidence},
-                virtual_evidence=None,
-                elimination_order=elimination_order,
-                joint=joint,
-                show_progress=show_progress,
-            )
+    # ✅ Step 1.5: Check for invalid query variables
+    invalid_vars = [var for var in variables if var not in self.model.nodes()]
```

Comment:
It would be better to put it together with `if not variable` as an else statement.